diff --git a/source/js/mouse-highlight.js b/source/js/mouse-highlight.js
index b9fd76e..c73284e 100644
--- a/source/js/mouse-highlight.js
+++ b/source/js/mouse-highlight.js
@@ -178,6 +178,16 @@ sitecues.def('mouse-highlight', function(mh, callback) {
 			collection = null;
 		}
 
+		function isContainedBy(item, container) {
+			while (item) {
+				if ($(container).has(item).length) {
+					return true;
+				}
+				item = item.parentNode;
+			}
+			return false;
+		}
+
 		mh.update = function(event) {
 			// break if highlight is disabled
 			if (!mh.enabled) return;
@@ -190,24 +200,35 @@ sitecues.def('mouse-highlight', function(mh, callback) {
 			if (!document.hasFocus())
 				return;
 
-			if (event.target !== mh.target) {
+			// Don't check picker more than we need to:
+			// - If we're on the same element as last time, we don't need a new check
+			// - If we're on an element within the current highlight, we don't need a new check
+			if (!mh.picked || (event.target !== mh.target && !isContainedBy(event.target, mh.picked))) {
+
 				// hide highlight for picked element
 
-				if (mh.picked) mh.hide(mh.picked);
+				if (mh.picked) {
+					mh.hide(mh.picked);
+					mh.picked = null;
+				}
 
 				// save target element
 				mh.target = event.target;
 
-				// save picked element
-				mh.picked = picker.find(event.target);
-
-				// show highlight for picked element
-				if (mh.picked && mh.picked.length) {
-					mh.timer && clearTimeout(mh.timer);
-					mh.timer = setTimeout(function() {
-						mh.show(mh.picked);
-					}, 100);
-				}
+				mh.pickerTimer && clearTimeout(mh.pickerTimer);
+				mh.highlightTimer && clearTimeout(mh.highlightTimer);
+				mh.pickerTimer = setTimeout(function() {
+					// save picked element
+					mh.picked = picker.find(event.target);
+
+					// show highlight for picked element
+					if (mh.picked && mh.picked.length) {
+						mh.highlightTimer && clearTimeout(mh.highlightTimer);
+						mh.highlightTimer = setTimeout(function() {
+							mh.show(mh.picked);
+						}, 1);
+					}
+				}, 20);
 			}
 
 		}
diff --git a/source/js/mouse-highlight/picker.js b/source/js/mouse-highlight/picker.js
index 44a7696..150ba58 100644
--- a/source/js/mouse-highlight/picker.js
+++ b/source/js/mouse-highlight/picker.js
@@ -34,6 +34,7 @@ sitecues.def('mouse-highlight/picker', function(picker, callback) {
         'false'    : 'f'
     }
 
+	picker.kMaxAncestorsToCheck = 7;
 	sitecues.use('jquery', 'style', 'mouse-highlight/roles', function($, styles, roles) {
 
 		/*
@@ -41,7 +42,7 @@ sitecues.def('mouse-highlight/picker', function(picker, callback) {
 		 *
 		 * @param hover The element the mouse is hovering over
 		 */
-		picker.find = function find(hover) {
+		picker.find = function find(hover, depth) {
 			var e = hover instanceof $ ? hover : $(hover);
 			if (e.is('body')) {
 				// We're at the body, we're done.
@@ -78,7 +79,10 @@ sitecues.def('mouse-highlight/picker', function(picker, callback) {
 				return e;
 			}
 			// No candidates
-			return picker.find(e.parent());
+			depth = depth ? depth + 1 : 1;
+			if (depth >= picker.kMaxAncestorsToCheck)
+				return null;
+			return picker.find(e.parent(), depth ? depth + 1 : 1);
 		};
 
 		/*
diff --git a/source/js/mouse-highlight/roles.js b/source/js/mouse-highlight/roles.js
index 83ecd00..749d8ae 100644
--- a/source/js/mouse-highlight/roles.js
+++ b/source/js/mouse-highlight/roles.js
@@ -177,6 +177,7 @@ sitecues.def('mouse-highlight/roles', function(role, callback) {
 				'region'],
 			sitecues: [
 				'container'],
+			shouldContainText: true,
 			shouldHaveChildren: true,
 			canHighlight: true
 		},
